include/libtorrent/bt_peer_connection.hpp

```diff
@@ -402,6 +397,7 @@ namespace libtorrent
 		// used during an encrypted handshake then moved
 		// into m_enc_handler if rc4 encryption is negotiated
 		// otherwise it is destroyed when the handshake completes
+		// TODO: 2 this could probably be a std::unique_ptr
```

Comment:
I checked this before and it has to be a shared_ptr because two references to it are held in `m_enc_handler`, one for encrypting outgoing traffic and one for decrypting incoming.